os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -198,18 +198,25 @@ void binary_manager_add_binlist(FAR struct tcb_s *tcb)
 {
 	int bin_idx;
 
-	if (tcb->group) {
-		bin_idx = tcb->group->tg_binidx;
-		if (bin_idx > 0) {
-			/* Add a tcb to a head of list */
-			tcb->bin_blink = NULL;
-			if (tcb->sched_priority > BM_PRIORITY_MAX) {
+	bin_idx = tcb->group->tg_binidx;
+
+	/* A binary index, bin_idx is greater than 0 only if tcb is a thread of user binary.
+	 * Add tcb to the binary list for fault recovery in case of threads of user binary.
+	 * Otherwise, it is unnessasary for idle or kernel threads. */
+	if (bin_idx > 0) {
```

Comment:
How about changing like below? Then we can avoid the code duplication and minimize the pointer operation which to get BIN_XXXLIST. Every BIN_XXXLIST(bin_idx) is a pointer operation, and it takes some time.
```
struct tcb_s *list;
if (bin_idx > 0) {
	if (tcb->sched_priority > BM_PRIORITY_MAX) {
		list = BIN_RTLIST(bin_idx);
	} else {
		list = BIN_NRTLIST(bin_idx);
	}

	if (list) {
		tcb->bin_flink = list;
		list->bin_blink = tcb;
	}
	list = tcb;
}
```

Reply:
`list = tcb;` is wrong.

Reply:
Okay. Right. Let's ignore this.